apps/src/code-studio/components/progress/LessonLockDialog.jsx

```diff
@@ -11,264 +10,287 @@ import SectionSelector from './SectionSelector';
-                      lockStatus === LockStatus.Locked && styles.selectedCell
-                    ]}
-                  >
-                    <input
-                      type="radio"
-                      name={index}
-                      value={LockStatus.Locked}
-                      checked={lockStatus === LockStatus.Locked}
-                      onChange={this.handleRadioChange}
-                    />
-                  </td>
-                  <td
-                    style={[
-                      styles.tableCell,
-                      styles.radioCell,
-                      lockStatus === LockStatus.Editable && styles.selectedCell
-                    ]}
-                  >
-                    <input
-                      type="radio"
-                      name={index}
-                      value={LockStatus.Editable}
-                      checked={lockStatus === LockStatus.Editable}
-                      onChange={this.handleRadioChange}
-                    />
-                  </td>
-                  <td
-                    style={[
-                      styles.tableCell,
-                      styles.radioCell,
-                      lockStatus === LockStatus.ReadonlyAnswers &&
-                        styles.selectedCell
-                    ]}
-                  >
-                    <input
-                      type="radio"
-                      name={index}
-                      value={LockStatus.ReadonlyAnswers}
-                      checked={lockStatus === LockStatus.ReadonlyAnswers}
-                      onChange={this.handleRadioChange}
-                    />
-                  </td>
-                </tr>
-              ))}
-            </tbody>
-          </table>
-        </div>
-        <div style={styles.buttonContainer}>
-          <button
-            type="button"
-            style={progressStyles.baseButton}
-            onClick={this.props.handleClose}
-          >
-            {commonMsg.dialogCancel()}
-          </button>
-          <button
-            type="button"
-            style={[progressStyles.blueButton, hiddenUnlessSelectedSection]}
-            onClick={this.handleSave}
-            disabled={this.props.saving}
-          >
-            {this.props.saving ? commonMsg.saving() : commonMsg.save()}
-          </button>
-        </div>
-      </BaseDialog>
-    );
-  }
+        <thead>
+          <tr>
+            <th style={styles.headerRow}>{commonMsg.student()}</th>
+            <th style={styles.headerRow}>{commonMsg.locked()}</th>
+            <th style={styles.headerRow}>{commonMsg.editable()}</th>
+            <th style={styles.headerRow}>{commonMsg.answersVisible()}</th>
+          </tr>
+        </thead>
+        <tbody>
+          {lockStatuses.map(({name, lockStatus}, index) =>
+            renderStudentRow(index, name, lockStatus)
+          )}
+        </tbody>
+      </table>
+    </>
+  );
+
+  const renderStudentRow = (index, name, lockStatus) => (
```

Comment:
I slightly prefer splitting components out into separate components rather than render functions, just because it's easier to read and to test, no need to change anything here, just sharing an opinon!

Reply:
Sure we can chat about it! I don't think there are hard and fast rules on this and sometimes we'll just have to use our judgement. I took a skim through https://frontarm.com/james-k-nelson/how-should-i-separate-components/ which has some reasonable points. Based on the article I would put `renderStudentRow` in the View Components ("Cards and list items can be factored out of lists")

Reply:
Here's another article about it https://medium.com/front-end-weekly/react-quick-tip-split-up-your-render-method-5eb30e432e2